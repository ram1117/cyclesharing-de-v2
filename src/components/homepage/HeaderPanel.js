import { useDispatch } from 'react-redux';
import { setLangPack } from '../../redux/language/languageSlice';

const HeaderPanel = () => {
  const dispatch = useDispatch();
  return (
    <div className="header-bar biege">
      <h1 className="logo">Cycle Share</h1>
      <div className="lang-buttons-panel">
        <button
          className="lang-buttons buttons"
          type="button"
          onClick={() => {
            dispatch(setLangPack('EN'));
          }}
        >
          EN
        </button>
        <button
          className="lang-buttons buttons"
          type="button"
          onClick={() => {
            dispatch(setLangPack('DE'));
          }}
        >
          DE
        </button>
      </div>
    </div>
  );
};

export default HeaderPanel;
