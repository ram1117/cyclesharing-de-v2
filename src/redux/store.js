import { configureStore } from '@reduxjs/toolkit';
import logger from 'redux-logger';
import citiesReducer from './homepage/cititesSlice';
import detailSliceReducer from './detailspage/detailSlice';
import languageSliceReducer from './language/languageSlice';

const store = configureStore({
  reducer: {
    cities: citiesReducer,
    detail: detailSliceReducer,
    language: languageSliceReducer,
  },
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(logger),
});

export default store;
