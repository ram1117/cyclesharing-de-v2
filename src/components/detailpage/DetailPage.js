import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useParams } from 'react-router-dom';
import {
  fetchDetails,
  resetDetails,
} from '../../redux/detailspage/detailSlice';
import HeaderPanel from '../homepage/HeaderPanel';

const DetailPage = () => {
  const { nwId } = useParams();
  const detail = useSelector((state) => state.detail.detail);
  const status = useSelector((state) => state.detail.status);
  const lang = useSelector((state) => state.language.langPack);
  const dispatch = useDispatch();
  useEffect(() => {
    if (status === 'loading') {
      dispatch(fetchDetails(nwId));
    }
    return function cleanup() {
      dispatch(resetDetails());
    };
  }, [nwId, status, dispatch]);

  return (
    <>
      <header>
        <HeaderPanel />
      </header>
      <div className="provider-details beige">
        <h2>{detail.city}</h2>
        <h3>{detail.name}</h3>
        <h3>{detail.company}</h3>
      </div>

      <div className="location-details beige">
        {detail.stations.length === 0 && <h2>{lang.nostation}</h2>}
        {detail.stations.map((station) => (
          <div
            className="station-detail beige"
            key={station.id}
            data-testid="stations"
          >
            <h3>
              {lang.station}
              {station.name}
            </h3>
            <p>
              {lang.freebikes}
              <span className="free-qty">{` ${station.free_bikes}`}</span>
            </p>
            <p>
              {lang.emptyslots}
              <span className="empty-qty">
                {station.empty_slots === null ? ' N/A' : station.empty_slots}
              </span>
            </p>
            <input
              className="coordinates"
              readOnly
              value={`${station.latitude} , ${station.longitude}`}
            />
          </div>
        ))}
      </div>
    </>
  );
};

export default DetailPage;
