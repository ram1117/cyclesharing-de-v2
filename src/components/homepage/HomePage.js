import { useEffect, useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { Link } from 'react-router-dom';
import { fetchData } from '../../redux/homepage/cititesSlice';
import HeaderPanel from './HeaderPanel';

const HomePage = () => {
  const { cities, status } = useSelector((state) => state.cities);
  const [filterCities, setFilterCities] = useState([]);
  const dispatch = useDispatch();
  useEffect(() => {
    if (status === 'loading') {
      dispatch(fetchData());
    }
    if (status === 'completed') {
      setFilterCities(cities);
    }
  }, [cities, status, dispatch]);

  const list = useSelector((state) => state.cities.cityCapitals);
  const lang = useSelector((state) => state.language.langPack);
  const sortlist = [...list].sort();

  const handleClick = (event) => {
    const char = event.target.innerText;
    if (char === 'All' || char === 'Alle') {
      setFilterCities(cities);
    } else {
      const newArr = cities.filter(
        (cityObj) => cityObj.city.charAt(0) === char,
      );
      setFilterCities(newArr);
    }
  };

  return (
    <>
      <header>
        <HeaderPanel />
      </header>
      <div className="total-stats">
        <h2>{lang.countryHeader}</h2>
        <h3>{lang.mainSubHeader}</h3>
        <h4 data-testid="total">
          {lang.cities}
          <span>{` ${cities.length}`}</span>
        </h4>
      </div>
      <div className="button-pane">
        <button
          type="button"
          className="alphabet-btns buttons"
          onClick={handleClick}
        >
          {lang.all}
        </button>
        {sortlist.map((char) => (
          <button
            type="button"
            className="alphabet-btns buttons"
            key={char}
            onClick={handleClick}
            data-testid="alphabets"
          >
            {char}
          </button>
        ))}
      </div>
      {status === 'loading' && <h4>{lang.loading}</h4>}
      <div className="cities-wrapper">
        {filterCities.map((cityObj) => (
          <Link
            className="city-link"
            to={`details/${cityObj.id}`}
            key={cityObj.id}
          >
            <div
              id="city-tile"
              className="city-tile"
              data-href={cityObj.href}
              data-testid="cities"
            >
              <h3>{cityObj.city}</h3>
              <p>
                {lang.provider}
                <br />
                <span>{cityObj.name}</span>
              </p>
              <p>
                Lat:
                {` ${cityObj.latitude}`}
              </p>
              <p>
                Long:
                {` ${cityObj.longitude}`}
              </p>
            </div>
          </Link>
        ))}
      </div>
    </>
  );
};

export default HomePage;
