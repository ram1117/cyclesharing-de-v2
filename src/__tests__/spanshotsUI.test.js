import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import HomePage from '../components/homepage/HomePage';
import DetailPage from '../components/detailpage/DetailPage';
import HeaderPanel from '../components/homepage/HeaderPanel';
import store from '../redux/store';

describe('snapshot testing components', () => {
  it('testing homepage snapshot', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <HomePage />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('testing detailpage snapshot', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <DetailPage />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('testing headerpanel snapshot', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <HeaderPanel />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
