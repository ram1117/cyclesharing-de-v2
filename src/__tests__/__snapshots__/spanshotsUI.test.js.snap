// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing components testing detailpage snapshot 1`] = `
Array [
  <header>
    <div
      className="header-bar biege"
    >
      <h1
        className="logo"
      >
        Cycle Share
      </h1>
      <div
        className="lang-buttons-panel"
      >
        <button
          className="lang-buttons buttons"
          onClick={[Function]}
          type="button"
        >
          EN
        </button>
        <button
          className="lang-buttons buttons"
          onClick={[Function]}
          type="button"
        >
          DE
        </button>
      </div>
    </div>
  </header>,
  <div
    className="provider-details beige"
  >
    <h2 />
    <h3 />
    <h3 />
  </div>,
  <div
    className="location-details beige"
  >
    <h2>
      No station data found...
    </h2>
  </div>,
]
`;

exports[`snapshot testing components testing headerpanel snapshot 1`] = `
<div
  className="header-bar biege"
>
  <h1
    className="logo"
  >
    Cycle Share
  </h1>
  <div
    className="lang-buttons-panel"
  >
    <button
      className="lang-buttons buttons"
      onClick={[Function]}
      type="button"
    >
      EN
    </button>
    <button
      className="lang-buttons buttons"
      onClick={[Function]}
      type="button"
    >
      DE
    </button>
  </div>
</div>
`;

exports[`snapshot testing components testing homepage snapshot 1`] = `
Array [
  <header>
    <div
      className="header-bar biege"
    >
      <h1
        className="logo"
      >
        Cycle Share
      </h1>
      <div
        className="lang-buttons-panel"
      >
        <button
          className="lang-buttons buttons"
          onClick={[Function]}
          type="button"
        >
          EN
        </button>
        <button
          className="lang-buttons buttons"
          onClick={[Function]}
          type="button"
        >
          DE
        </button>
      </div>
    </div>
  </header>,
  <div
    className="total-stats"
  >
    <h2>
      Germany
    </h2>
    <h3>
      Eco-Friendly Mobility
    </h3>
    <h4
      data-testid="total"
    >
      Cities:
      <span>
         0
      </span>
    </h4>
  </div>,
  <div
    className="button-pane"
  >
    <button
      className="alphabet-btns buttons"
      onClick={[Function]}
      type="button"
    >
      All
    </button>
  </div>,
  <h4>
    loading...
  </h4>,
  <div
    className="cities-wrapper"
  />,
]
`;
